src/compile/legend/encode.ts

```diff
@@ -68,7 +68,11 @@ export function symbols(fieldDef: FieldDef<string>, symbolsSpec: any, model: Uni
   if (channel !== OPACITY) {
     const opacityDef = model.encoding.opacity;
     if (isValueDef(opacityDef)) {
-      symbols.opacity = {value: opacityDef.value};
+      if (hasConditionValueDef(opacityDef)) {
```

Comment:
1) Please extract method `getOpacityValue` so we can reuse below. 

2) How should we deal with `{condition: { value: ... }, field: ..., ...}` and `{condition: { field: ... }, value: ..., ...}`?  I think two of these should be consistent but given the current code, the latter will get fixed opacity. 
  - If you deal with the case where condition is a fielddef too, maybe you do not need to add the new `hasConditionValueDef`.